[ALLUXIO-2741] Convert function mac_hfs_provision_sectors in "bin/alluxio-mount.sh" to Java

LuQQiu opened the pull request:
https://alluxio.atlassian.net/browse/ALLUXIO-2741

It seems that the request size may be very big in byte, so I use BigDecimal to do all the calculation.
The calculation result is a little different from the previous one in alluxio-mount.sh because of the round up way.

  n_100g=$(printf "%.0f" $(echo "scale=4; ${request_size}/107374182400 + 0.5" | bc)) # round up
  n_gb=$(printf "%.0f" $(echo "scale=4; ${request_size}/1073741824 + 0.5" | bc)) # round up
  n_kb=$(printf "%.0f" $(echo "scale=4; ${request_size}/1024 + 0.5" | bc)) #round up

  BigDecimal memSizeKB = memSize.divide(sizeKB, 4, RoundingMode.HALF_UP); // round up
  BigDecimal memSizeGB = memSize.divide(sizeGB, 4, RoundingMode.HALF_UP); // round up
  BigDecimal memSize100GB = memSize.divide(hundredGB, 8, RoundingMode.HALF_UP); // round up

LuQQiu commented:
alluxio-bot, check this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/14655/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/14656/
Test PASSed.

apc999 commented on core/common/src/main/java/alluxio/util/HFSUtils.java:
```diff
@@ -0,0 +1,101 @@
+    // Metadata zone size is estimated by using specifications noted at
+    // http://dubeiko.com/development/FileSystems/HFSPLUS/tn1150.html#MetadataZone,
+    // which is slightly larger than actually metadata zone size created by hdiuitl command.
+    BigDecimal memSize = new BigDecimal(requestSize);
```
any reason to use BigDecimal rather than long?

apc999 commented on core/common/src/main/java/alluxio/util/HFSUtils.java:
```diff
@@ -0,0 +1,101 @@
+import java.math.RoundingMode;
+
+/**
+ * This util class is called in alluxio-mount.sh to convert the memory size to number of sectors.
```
Add comment that util methods here are for calculating space for Mac OS X HFS+?

LuQQiu commented:
Thanks for your suggestions!

I changed from BigDecimal to Double which is able to represent TB in Bytes and maintain precision in division operations. 

The return type I changed to long value.

The result is same as I use BigDecimal but the codes look more reasonable and readable.

PTAL @apc999

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/14964/<h2>Failed Tests: <span class='status-failure'>8</span></h2><h3><a name='org.alluxio:alluxio-core-common' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport'>org.alluxio:alluxio-core-common</a>: <span class='status-failure'>7</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest100GB/'><strong>alluxio.util.HFSUtilsTest.getSectorTest100GB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest512GB/'><strong>alluxio.util.HFSUtilsTest.getSectorTest512GB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest512/'><strong>alluxio.util.HFSUtilsTest.getSectorTest512</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest20/'><strong>alluxio.util.HFSUtilsTest.getSectorTest20</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestGB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestGB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestMB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestMB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestTB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestTB</strong></a></li></ul><h3><a name='org.alluxio:alluxio-tests' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-tests/testReport'>org.alluxio:alluxio-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14964/org.alluxio$alluxio-tests/testReport/alluxio.hadoop/FileSystemAclIntegrationTest/parentDirectoryPermissionForUfs/'><strong>alluxio.hadoop.FileSystemAclIntegrationTest.parentDirectoryPermissionForUfs</strong></a></li></ul>
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/14965/<h2>Failed Tests: <span class='status-failure'>7</span></h2><h3><a name='org.alluxio:alluxio-core-common' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport'>org.alluxio:alluxio-core-common</a>: <span class='status-failure'>7</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest100GB/'><strong>alluxio.util.HFSUtilsTest.getSectorTest100GB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest512GB/'><strong>alluxio.util.HFSUtilsTest.getSectorTest512GB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest512/'><strong>alluxio.util.HFSUtilsTest.getSectorTest512</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTest20/'><strong>alluxio.util.HFSUtilsTest.getSectorTest20</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestGB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestGB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestMB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestMB</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Alluxio-Pull-Request-Builder/14965/org.alluxio$alluxio-core-common/testReport/alluxio.util/HFSUtilsTest/getSectorTestTB/'><strong>alluxio.util.HFSUtilsTest.getSectorTestTB</strong></a></li></ul>
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/14991/
Test PASSed.
